File: app/src/main/java/xyz/xyz0z0/androidadvanced/MainActivity.java
package xyz.xyz0z0.androidadvanced;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;
import xyz.xyz0z0.androidadvanced.c02.Chapter02Activity;
import xyz.xyz0z0.androidadvanced.c02.Student;

public class MainActivity extends AppCompatActivity {

    private Button btnC02;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        btnC02 = findViewById(R.id.btnC02);
        btnC02.setOnClickListener(new View.OnClickListener() {
            @Override public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, Chapter02Activity.class);
                intent.putExtra("name", "Jack");
                intent.putExtra("age", 20);
                intent.putExtra("student", new Student("Li"));
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/xyz/xyz0z0/androidadvanced/MainActivity.kt
package xyz.xyz0z0.androidadvanced

import android.os.Bundle
import android.os.SystemClock
import android.util.Log
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private lateinit var btnC02: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btnC02 = findViewById(R.id.btnC02)

        val threadTest = ThreadTest()


        btnC02.setOnClickListener(View.OnClickListener { // Intent intent = new Intent(MainActivity.this, Chapter02Activity.class);
            // intent.putExtra("name", "Jack");
            // intent.putExtra("age", 20);
            // intent.putExtra("student", new Student("Li"));
            // startActivity(intent);
//            val s = "2303"
//            Log.d("cxg", "" + s.toTest())
            Log.d("cxg-time",""+SystemClock.uptimeMillis())
            Thread(Runnable {
                for (i in 1..5) {
                    threadTest.next("111")
                }
            }).start()
            Thread(Runnable {
                for (i in 1..5) {
                    threadTest.enqueue("222")
                }
            }).start()
        })


    }
}